### DISCUSSION TITLE
The client does not start on the mac

### DISCUSSION THREAD

**Discussion by iEvkaa** [rustdesk/rustdesk#8593] [Q&A] [answered]
Hi! We use rustdesk pro. We are trying to configure our clients. Everything works fine on Windows. However, the dmg file does not open on mac. the error is "unable to open". What could be the problem?
P.s To be honest, I do not know what to write in the Bundle identifier, so I write com.namemycompany.rustdesk

**Comment by iEvkaa** [comment]
I'll add my question a little bit.. There are no problems with the versions downloaded from the official website. However, we bought the pro version because we want to connect to our employees. To do this, we need to transfer the binary file to the employee to run. For convenience, I do not want the employee to enter the data himself to connect to our server. That's why we use client assembly via the web interface of our server. And this client does not run on the mac. If the custom client does not work, then I would like to do something like naming the client as in Windows. However, I have not found a solution yet.

**Reply by 21pages** [reply]
Can you take a screenshot of your custom client configuration? Remember to hide private information.

**Reply by iEvkaa** [reply]
I have hidden the personal information labeled "screenshot". There are no settings in this client, because I thought that maybe the problem was in them, but even a clean client does not start for me. I tried to pull the app file from the official dmg and paste it into my dmg. However, not successfully 
![image](https://github.com/rustdesk/rustdesk/assets/163337100/4d0f3fd8-2d72-468e-a250-82363ae679e0)

**Reply by 21pages** [reply]
What's the version of your custom mac client? `the versions downloaded from the official website` is 1.2.7?

![1720596281831](https://github.com/rustdesk/rustdesk/assets/14891774/32c3246b-f3a7-4ab3-976c-f1e048101e96)

**Reply by iEvkaa** [reply]
If I understand correctly, then my client's version is 1.2.6. that's what it says in the "Info.plist" file.I download the same version from the website. Now I'm thinking that maybe I'm building another version, but I don't know how to check it yet

**Reply by iEvkaa** [reply]
Version for Windows 1.2.4

**Reply by iEvkaa** [reply]
I don't understand why this is happening. I update my server regularly. Right now I'm upgrading from 1.3.10 to 1.3.11

**Reply by 21pages** [reply]
You can find the client version from Settings -> About

When `unable to open` happens? 
* Click on the downloaded custom client dmg
* After move it to applications, click it on LaunchPad
* After install, click it on LaunchPad

**Reply by iEvkaa** [reply]
"After move it to applications, click it on LaunchPad" at this point I get an error

**Reply by 21pages** [reply]
Does your app name or com.xxx.xxx contain non-English letters?

**Reply by iEvkaa** [reply]
No, I only use English letters.

**Reply by iEvkaa** [reply]
com.mycompany.subdomain, I'm writing like this, however I'm not sure what will work. I tried changing it to com.carriez.rustdesk, but it didn't help either

**Reply by iEvkaa** [reply]
I am currently testing on this configuration
Macbook Air 2020 M1-chip
MacOS Sonoma 14.5

**Reply by 21pages** [reply]
Upgraded to 14.5, still can't reproduce. Could you send your client dmg without real server config to my email? `c3VuYm9lYXN5QGdtYWlsLmNvbQo=`, base64.

**Reply by 21pages** [reply]
Could you show the `unable to open` dialog? is it because of `unidentified developer` ?
![image](https://github.com/rustdesk/rustdesk/assets/14891774/a70bca38-b666-47af-8588-b8a152096a97)

**Reply by iEvkaa** [reply]
Unfortunately my screenshot is only in Russian
![image](https://github.com/rustdesk/rustdesk/assets/163337100/78853480-6a98-48b4-bb73-e22921c59ff3)

**Reply by 21pages** [reply]
It works on my mac.
![image](https://github.com/rustdesk/rustdesk/assets/14891774/86c7e145-eaa9-49f0-b10e-058a3eabe40b)

Could you confirm again that these 2 programs can be opened on your mac?

https://github.com/rustdesk/rustdesk/releases/download/1.2.6/rustdesk-1.2.6-aarch64.dmg
https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.7-aarch64.dmg

>  I tried changing it to com.carriez.rustdesk, but it didn't help either

You rebuild one but still not work?

**Reply by iEvkaa** [reply]
It's very strange. Both clients that you dropped off here are not launching. And the client that I downloaded from the official website right now has started. 
![image](https://github.com/rustdesk/rustdesk/assets/163337100/148e88bf-6270-49f0-889f-e00166b1fa20)

**Reply by iEvkaa** [reply]
Apparently it's my mac

**Reply by 21pages** [reply]
The link in the red circle is same as the first link.

**Reply by iEvkaa** [reply, marked as answer]
Guys, thank you very much! I figured out the problem. Now I will try to explain in detail, maybe it will help someone. Personally, I don't have a mac, so all the testing took place on my friend's mac. I threw him the mac clients in a telegram. Apparently he's doing something with the metadata, so the clients didn't start. And the very first client that I posted on the network did not start, apparently because I renamed it Windows. Now I have made a client that I have placed on the corporate network and everything works there. I'm sorry for distracting you with my question!

**Reply by itzhuk** [reply]
> Guys, thank you very much! I figured out the problem. Now I will try to explain in detail, maybe it will help someone. Personally, I don't have a mac, so all the testing took place on my friend's mac. I threw him the mac clients in a telegram. Apparently he's doing something with the metadata, so the clients didn't start. And the very first client that I posted on the network did not start, apparently because I renamed it Windows. Now I have made a client that I have placed on the corporate network and everything works there. I'm sorry for distracting you with my question!

You are a holy man! I don’t know if we could figure it out on our own, and how long it would take. Thank you for leaving your comment. 

After transferring a file via Telegram, it does not launch on Macos. I sent the same file by mail - and everything is fine.
